src/tasks/minions/farmingActivity.ts

```diff
@@ -0,0 +1,426 @@
+						await user.removeGP(200 * alivePlants);
+					}
+
+					harvestXp = 0;
+				}
+				if (plantToHarvest.givesLogs && chopped) {
+					if (!plantToHarvest.outputLogs) return;
+					if (!plantToHarvest.woodcuttingXp) return;
+					const amountOfLogs = rand(5, 10);
+					loot[plantToHarvest.outputLogs] = amountOfLogs * alivePlants;
+
+					woodcuttingXp += alivePlants * amountOfLogs * plantToHarvest.woodcuttingXp;
+					wcStr = ` You also received ${woodcuttingXp.toLocaleString()} Woodcutting XP.`;
+
+					harvestXp = 0;
+				} else if (plantToHarvest.givesCrops && chopped) {
+					if (!plantToHarvest.outputCrop) return;
+					loot[plantToHarvest.outputCrop] = cropYield * alivePlants;
+
+					harvestXp = cropYield * alivePlants * plantToHarvest.harvestXp;
+				}
+			}
+
+			if (quantity > patchType.lastQuantity) {
+				loot[6055] = (quantity - patchType.lastQuantity) * 3; // weeds
+				rakeXp = (quantity - patchType.lastQuantity) * 3 * 4;
+				rakeStr += ` ${rakeXp} XP for raking, `;
+			}
+
+			farmingXpReceived = plantXp + harvestXp + checkHealthXp + rakeXp;
+			let deathStr = '';
+			if (quantityDead > 0) {
+				deathStr = ` During your harvest, you found that ${quantityDead}/${patchType.lastQuantity} of your plants died.`;
+			}
+
+			if (planting) {
+				plantingStr = `${user}, ${user.minionName} finished planting ${quantity}x ${plant.name} and `;
+			} else plantingStr = `${user}, ${user.minionName} finished `;
+
+			let str = `${plantingStr}harvesting ${patchType.lastQuantity}x ${
+				plantToHarvest.name
+			}.${deathStr}${payStr}\n\nYou received ${plantXp.toLocaleString()} XP for planting, ${rakeStr}${harvestXp.toLocaleString()} XP for harvesting, and ${checkHealthXp.toLocaleString()} XP for checking health for a total of ${farmingXpReceived.toLocaleString()} Farming XP.${wcStr}\n`;
+
+			bonusXP += Math.floor(farmingXpReceived * bonusXpMultiplier);
+
+			if (bonusXP > 0) {
+				str += `You received an additional ${bonusXP.toLocaleString()} bonus XP from your farmer's outfit.`;
+			}
+
+			await user.addXP(SkillsEnum.Farming, Math.floor(farmingXpReceived + bonusXP));
+			await user.addXP(SkillsEnum.Woodcutting, Math.floor(woodcuttingXp));
+
+			const newFarmingLevel = user.skillLevel(SkillsEnum.Farming);
+			const newWoodcuttingLevel = user.skillLevel(SkillsEnum.Woodcutting);
+
+			if (newFarmingLevel > currentFarmingLevel) {
+				str += `\n\n${user.minionName}'s Farming level is now ${newFarmingLevel}!`;
+			}
+
+			if (newWoodcuttingLevel > currentWoodcuttingLevel) {
+				str += `\n\n${user.minionName}'s Woodcutting level is now ${newWoodcuttingLevel}!`;
+			}
+
+			let tangleroot = false;
+			if (plantToHarvest.seedType === 'hespori') {
+				await user.incrementMonsterScore(Monsters.Hespori.id);
+				const hesporiLoot = Monsters.Hespori.kill(1, { farmingLevel: currentFarmingLevel });
+				loot = hesporiLoot;
+				if (Object.keys(loot).length > 2) {
+					tangleroot = true;
+				}
```

Comment:
this is not a good way to check. even if its not possible and technically works, its hacky.